guides/NEPTUNE.md

```diff
@@ -0,0 +1,101 @@
+Migrating from Neo4j to AWS Neptune
+=======================
+
+A developer guide for moving the CMD services from using Neo4j to AWS Neptune.
+
+### Introduction to Neptune
+
+- Neptune is a graph database that is provided as a managed AWS service. This saves a lot of the operational overhead we had with Neo4j, and provides clustering capabilities out of the box.
+- Neptune supports both Gremlin and SPARQL as graph query languages. Our implementation is using Gremlin.
+- A Neptune cluster has a primary instance for read and write operations, and up to 15 read replica instances. When connecting to Neptune, the cluster endpoint will connect to the primary instance. There is also a reader endpoint which will connect to one of the available read only instances.
+- Neptune mostly complies with Gremlin standards but there are some caveats. This is the reason we cannot simply run another Gremlin compliant graph database locally. [See here](https://docs.aws.amazon.com/neptune/latest/userguide/feature-overview-standards-compliance.html) for more details on the standard compliance.
+- For more information on Neptune refer to the [Neptune docs](https://docs.aws.amazon.com/neptune/latest/userguide/intro.html)
+
+### Local development - How do I run the CMD services against Neptune?
+
+Neo4j runs within a docker container as part of the [dp-compose](https://github.com/ONSdigital/dp-compose) project. Running a local instance of Neptune is not possible, so we must use a test instance hosted in AWS.
```

Comment:
Just a comment for the future: 
There are some dockerised graph DBs that support gremlin. (e.g. https://hub.docker.com/r/tinkerpop/gremlin-server/) - would we consider adding something like this to dp-compose to have a fully local environment in the future? Or maybe it's not worth it?

Reply:
It was attempted during the early phases of the Neptune migration but unfortunately did not work. This is due to Neptune not being fully compliant with the Gremlin standards. I touch on this on line 11, but maybe I can add some more to emphasise it?